packages/web/src/components/usdc-purchase-details-modal/USDCPurchaseDetailsModal.tsx

```diff
@@ -0,0 +1,247 @@
+import { useCallback } from 'react'
+
+import {
+  CommonState,
+  USDCPurchaseDetails,
+  chatActions,
+  chatSelectors,
+  formatUSDCWeiToUSDString,
+  useGetTrackById,
+  useInboxUnavailableModal,
+  useUSDCPurchaseDetailsModal
+} from '@audius/common'
+import {
+  HarmonyButton,
+  HarmonyButtonType,
+  IconArrow,
+  IconCart,
+  IconMessage,
+  Modal,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalTitle
+} from '@audius/stems'
+import moment from 'moment'
+import { useSelector, useDispatch } from 'react-redux'
+
+import { Icon } from 'components/Icon'
+import { Link } from 'components/link'
+import { DynamicTrackArtwork } from 'components/track/DynamicTrackArtwork'
+import { Text } from 'components/typography'
+import { UserNameAndBadges } from 'components/user-name-and-badges/UserNameAndBadges'
+import { useGoToRoute } from 'hooks/useGoToRoute'
+
+import styles from './USDCPurchaseDetailsModal.module.css'
+
+const { getCanCreateChat } = chatSelectors
+const { createChat } = chatActions
+
+const messages = {
+  by: 'By',
+  cost: 'Cost',
+  date: 'Date',
+  done: 'Done',
+  messageBuyer: 'Message Buyer',
+  purchasedBy: 'Purchased By',
+  purchaseDetails: 'Purchase Details',
+  saleDetails: 'Sale Details',
+  track: 'Track',
+  trackPurchased: 'Track Purchased',
+  visitTrack: 'Visit Track'
+}
+
+type ContentProps = {
+  purchaseDetails: USDCPurchaseDetails
+  onClose: () => void
+}
+
+const DetailSection = ({
+  children,
+  label
+}: {
+  children: React.ReactNode
+  label: string
+}) => (
+  <div className={styles.detailSection}>
+    <Text variant='label' size='large' color='neutralLight4'>
+      {label}
+    </Text>
+    {children}
+  </div>
+)
+
+const TrackLink = ({ id, onClick }: { id: number; onClick: () => void }) => {
+  const { data: track } = useGetTrackById({ id })
+  if (!track) return null
+  return (
+    <Link onClick={onClick} className={styles.link} to={track.permalink}>
+      <Text size='large' color='secondary'>
+        {track.title}
+      </Text>
+    </Link>
+  )
+}
+
+const PurchaseModalContent = ({ purchaseDetails, onClose }: ContentProps) => {
```

Comment:
these feel like they should be separate files to me

Reply:
You are probably right. I had thought they would be fairly small, but apparently not. The goal was to just split up some of the nested components without exporting them for use elsewhere. But I suppose we already have the pattern of nested components folders to keep files from getting too big.

Reply:
Alright nested everything in a components sub-folder. Slightly _less_ shared code than all being in a module together, but way more readable :-)